zad2/warriors.rb

```diff
@@ -1,12 +1,79 @@
-class Warrior
+class Character
+  attr_reader :level
+
   def initialize(name:, level:)
+    unless level.between? @@MIN_LEVEL, @@MAX_LEVEL 
+      raise ArgumentError, "Level should be in range [#@@MIN_LEVEL, #@@MAX_LEVEL]"
+    end
+
+    @name = name
+    @level = level
   end
 
   def strength
+    @level + roll_dice
+  end
+
+  def card
+    "#@name (lvl #@level)"
+  end
+
+  def to_s
+    @name
+  end
+
+  protected
+
+  @@MIN_LEVEL = 1
```

Comment:
to bardziej pasuje do stałej niż zmiennej klasowej. Zmienne klasowe można zmieniać, a min_level i max_level są z góry zdefiniowane.

Reply:
w sumie zmiennych klasowych raczej się nie używa na codzień w ruby
![](https://www.rubypigeon.com/images/ruby-you-can-but-dont.png)

Reply:
Poprawione :+1: